src/Orc.FileAssociation/Services/FileAssociationService.cs

```diff
@@ -7,44 +7,56 @@
 
 namespace Orc.FileAssociation
 {
+    using System;
     using System.Runtime.InteropServices;
     using Catel;
     using Catel.Logging;
+    using Microsoft.Win32;
 
     public class FileAssociationService : IFileAssociationService
     {
         private static readonly ILog Log = LogManager.GetCurrentClassLogger();
 
-        public void AssociateFilesWithApplication(string applicationName = null)
+        public void AssociateFilesWithApplication(ApplicationInfo applicationInfo)
         {
-            applicationName = applicationName.GetApplicationName();
+            string applicationName = null;
+            if (applicationInfo.Name.GetApplicationName() is not null)
```

Comment:
Store result in variable instead of calling it twice (we don't know how expensive the operation is)